src/util/OAuth2Util.js

```diff
@@ -95,6 +95,72 @@ define(['okta', './Enums', './Errors'], function (Okta, Enums, Errors) {
     }
   };
 
+  // Parse through the OAuth 'authParams' object to ensure the 'openid' scope is
+  // included (if required)
+  util.addOrRemoveOpenIdScope = function (authParams) {
+    if (!authParams.responseType) {
+      return;
+    }
+
+    //  Convert scope into an Array
+    var scope = Array.isArray(authParams.scopes) ? authParams.scopes : authParams.scopes.split(' ');
+
+    // Remove the 'openid' scope, as it is only required if an 'id_token' is requested
+    if (scope.includes('openid')) {
+      scope.splice(scope.indexOf('openid'), 1);
+    }
+
+    // Add the 'openid' scope
+    if (authParams.responseType.includes('id_token')) {
+      scope.push('openid');
+    }
+
+    return scope;
+  };
+
+  // Utility handlers for mapping convenience keys to OAuth params
+  util.getResponseType = function (options) {
+    var responseType = [];
+    if (options.getIdToken !== false) {
+      responseType.push('id_token');
+    }
+
+    if (options.getAccessToken) {
+      responseType.push('token');
+    }
+
+    return responseType;
+  };
+
+  util.transformShowSignInToGetTokensOptions = function(options, config = {}) {
+    // Override specific OAuth/OIDC values
+    var renderOptions = {
+      clientId: options.clientId,
```

Comment:
Per offline convo, if clientId is not defined, we stall and do nothing because of this logic:

https://github.com/okta/okta-signin-widget/blob/master/src/models/Settings.js#L210

https://github.com/okta/okta-signin-widget/blob/master/src/util/RouterUtil.js#L114

We need some better way to alert the developer that the clientId is required and was not provided.